tools/scripts/rebuildSoundLibraryManifest.rb

```diff
@@ -0,0 +1,382 @@
+        @warnings.push result
+      end
+      metadata_progress_bar.increment unless metadata_progress_bar.nil?
+    end
+) do |name|
+      # This is the parallel block.  This block should return a string to
+      # generate a warning and skip the sound, and a metadata Hash in
+      # the success case.
+      objects = sound_objects[name]
+
+      # Drop this sound if it is missing its JSON or MP3 components
+      if objects['json'].nil?
+        next "sound #{name} does not have a JSON file and was skipped."
+      elsif objects['mp3'].nil?
+        next "sound #{name} does not have a MP3 file and was skipped."
+      end
+
+      # Before we do anything else, check the last modify times on both files.
+      # If they are unchanged, just use the old metadata (+aliases).
+      # This saves us from actually downloading the files from S3
+      if previous_metadata[name].is_a?(Hash) &&
+          previous_metadata[name]['jsonLastModified'] == objects['json'].last_modified.to_s &&
+          previous_metadata[name]['mp3LastModified'] == objects['mp3'].last_modified.to_s
+        verbose "#{bold name} is unchanged, using old metadata"
+        next previous_metadata[name]
+      end
+
+      # Actually download the JSON from S3
+      begin
+        json_response = objects['json'].get
+        metadata = JSON.parse(json_response.body.read)
+      rescue Aws::Errors::ServiceError => service_error
+        next <<-WARN
+There was an error retrieving #{name}.json from S3:
+#{service_error}
+The sound has been skipped.
+        WARN
+      rescue JSON::JSONError => json_error
+        next <<-WARN
+There was an error parsing #{name}.json:
+#{json_error}
+The sound has been skipped.
+        WARN
+      end
+
+      # Verify that the parsed metadata contains all expected fields
+      next "sound #{name} is missing the 'name' attribute." unless metadata['name'].is_a?(String)
+      next "sound #{name} is missing the 'sound' attribute." unless metadata['time'].is_a?(Integer)
+
+      # Add last modification time for each file to the metadata so we can skip
+      # unmodified files in future manifest updates
+      metadata['jsonLastModified'] = objects['json'].last_modified.to_s
+      metadata['mp3LastModified'] = objects['mp3'].last_modified.to_s
+
+      # Record target version in the metadata, so environments (and projects)
+      # consistently reference the version they originally imported.
+      metadata['version'] = objects['mp3'].object.version_id
+
+      # Generate appropriate sourceUrl pointing to the sound library API
+      metadata['sourceUrl'] = "/api/v1/sound-library/#{metadata['version']}/#{name}.mp3"
+
+      verbose <<-EOS
+#{bold name} @ #{metadata['version']}
+#{JSON.pretty_generate metadata}
+      EOS
+
+      metadata
+    end
+    metadata_progress_bar.finish unless metadata_progress_bar.nil?
+    sound_metadata_by_name
+  end
+
+  # Given a metadata map, build the alias map
+  def build_alias_map(sound_metadata)
```

Comment:
I think there's nothing sound-specific in this method.  Can we somehow extract it and share it between the sound manifest and animation manifest scripts?